rust/src/domains/polars/frame/test.rs

```diff
@@ -39,3 +39,136 @@ fn test_margin() -> Fallible<()> {
 
     Ok(())
 }
+
+#[test]
+fn test_get_margin_max_partition_descriptors() -> Fallible<()> {
+    let lf_domain = LazyFrameDomain::new(vec![
+        SeriesDomain::new("A", AtomDomain::<i32>::default()),
+        SeriesDomain::new("B", AtomDomain::<i32>::default()),
+    ])?
+    .with_margin(
+        &["A"],
+        Margin::default()
+            .with_max_partition_length(10)
+            .with_max_partition_contributions(3),
+    )?;
+
+    assert_eq!(
+        lf_domain
+            .get_margin(BTreeSet::from(["A".to_string(), "B".to_string()]))
+            .max_partition_length,
+        Some(10)
+    );
+
+    assert_eq!(
+        lf_domain
+            .get_margin(BTreeSet::from(["B".to_string()]))
+            .max_partition_length,
+        None
+    );
+
+    assert_eq!(
+        lf_domain
+            .get_margin(BTreeSet::from(["A".to_string(), "B".to_string()]))
+            .max_partition_contributions,
+        Some(3)
+    );
+
+    assert_eq!(
+        lf_domain
+            .get_margin(BTreeSet::from(["B".to_string()]))
+            .max_partition_contributions,
+        None
+    );
+    Ok(())
+}
+
+#[test]
+fn test_get_margin_covering_small_to_large() -> Fallible<()> {
+    let lf_domain = LazyFrameDomain::new(vec![
+        SeriesDomain::new("A", AtomDomain::<i32>::default()),
+        SeriesDomain::new("B", AtomDomain::<i32>::default()),
+    ])?
+    .with_margin(
+        &["A"],
+        Margin::default()
+            .with_max_num_partitions(10)
+            .with_max_influenced_partitions(3),
+    )?
+    .with_margin(
+        &["B"],
+        Margin::default()
+            .with_max_num_partitions(10)
+            .with_max_influenced_partitions(3),
```

Comment:
Slightly stronger test if these are set to different values than the first margin. Not that it's likely, but if the code just used the first margin twice, these tests wouldn't catch the problem.